### PR TITLE
FileSystem level credential test

### PR DESCRIPTION
**Pull request by cloud-fan**
**Description of changes**

Improve the credential testing in Spark integration, to verify the credential at the low level `FileSystem`.

### CODE REVIEW

File: connectors/spark/etc/conf/server.properties [hunk 1]
```diff
@@ -1,6 +1,10 @@
 server.env=dev
 ## temp credential config for s3 (Multiple s3 config can be added by incrementing the index)
```
Comment (tdas) [inline connectors/spark/etc/conf/server.properties:2]: @ravivj-db I really think that server.properties should be loadable from the classpath as well. this is not a good set up. this file should be spark/test/resources

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java [hunk 2]
```diff
@@ -207,142 +156,77 @@ private SparkSession createSparkSessionWithCatalogs(String... catalogs) {
...
             List<String> partitionColumns,
-            boolean isManaged,
-            SparkSession session) throws IOException, ApiException {
+            boolean isManaged) throws IOException, ApiException {
         Integer partitionIndex1 = partitionColumns.indexOf("i");
         if (partitionIndex1 == -1) partitionIndex1 = null;
         Integer partitionIndex2 = partitionColumns.indexOf("s");
         if (partitionIndex2 == -1) partitionIndex2 = null;
 
-        // TODO: remove this when managed table is officially supported.
```
Comment (tdas) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:285]: why is this removed.

Comment (cloud-fan) [reply connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:285]: I no longer need to create managed tables to test credentials. I'll add managed table support once UC server is ready.

File: connectors/spark/etc/conf/server.properties (hunk 1 above)
Comment (ravivj-db) [reply connectors/spark/etc/conf/server.properties:2]: ACK.

**Review by jaceklaskowski — APPROVED** [review]
LGTM

**Comment by jaceklaskowski** [comment]
@cloud-fan Can you edit the description of the PR and remove the checkboxes so it won't clutter up `git log` (it's nice in the UI but terrible on command line)?

File: build.sbt [hunk 3]
```diff
@@ -337,6 +328,36 @@ lazy val spark = (project in file("connectors/spark"))
       "net.aichler" % "jupiter-interface" % JupiterKeys.jupiterVersion.value % Test,
       "io.delta" %% "delta-spark" % "4.0.0rc1" % Test,
     ),
+    excludeDependencies ++= Seq(
+      ExclusionRule("com.adobe.testing", "s3mock-junit5")
```
Comment (tdas) [inline build.sbt:332]: what is this for? can you leave a comment? who is bringing in this and why does it need to be excluded

File: connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala [hunk 4]
```diff
@@ -99,7 +99,6 @@ private class UCProxy extends TableCatalog {
       tablesApi.getTable(name + "." + ident.toString)
     } catch {
       case e: ApiException if e.getCode == 404 =>
-        e.printStackTrace()
```
Comment (tdas) [inline connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala:102]: @cloud-fan in your follow up PR can you set up some slf4j logging?

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/FakeFileSystem.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
+package io.unitycatalog.connectors.spark;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.*;
+import org.apache.hadoop.util.Progressable;
+
+import java.io.IOException;
+
+public class FakeFileSystem extends RawLocalFileSystem {
```
Comment (tdas) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/FakeFileSystem.java:9]: can you add class docs to explain the purpose of this class? 
Also, maybe rename this to "CredentialTestFileSystem" to make its purpose more intuitive than just "Fake"?

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java [hunk 6]
```diff
@@ -110,65 +98,53 @@ public void testDeltaPathTable() throws IOException, ApiException {
     }
 
     @Test
-    public void testManagedParquetTable() throws ApiException, IOException, ParseException {
+    public void testCredentialParquet() throws ApiException, IOException {
```
Comment (tdas) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:101]: why not just `testParquet()`. it is doing the full read/write test, not just credentials... isnt it?

Comment (cloud-fan) [reply connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:101]: There is already `testParquetReadWrite` test in this file. The difference of this new test is it uses s3 so that it tests credentials.

File: connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala (hunk 4 above)
Comment (cloud-fan) [reply connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala:102]: sure

**Review by tdas — APPROVED** [review]
